lib/FusionInventory/Agent/Task/Inventory/Win32/Users.pm

```diff
@@ -145,12 +157,18 @@ sub _getLastUser {
     }) or die "Can't open HKEY_LOCAL_MACHINE key: $EXTENDED_OS_ERROR";
 
     my $user =
+        encodeFromRegistry($machKey->{"SOFTWARE/Microsoft/Windows/CurrentVersion/Authentication/LogonUI/LastLoggedOnSAMUser"}) ||
         encodeFromRegistry($machKey->{"SOFTWARE/Microsoft/Windows/CurrentVersion/Authentication/LogonUI/LastLoggedOnUser"}) ||
         encodeFromRegistry($machKey->{"SOFTWARE/Microsoft/Windows NT/CurrentVersion/Winlogon/DefaultUserName"});
     return unless $user;
 
+    my ($domain) = $user =~ /^(.*)\\/;
```

Comment:
This just extract the domain name from the fully-qualified user name, instead of splitting in two distinct components. Additionally, it assumes the value uses the domain\user syntax: if it uses the alternative user@domain syntax, as demonstrated in issue #337, it won't work.